CDCgov/prime-reportstream
# Cle/6086 hl7 delimeters

This PR changes HLY MSH separator input to remove '#' where present. 

Test Steps:
1. Send an HL7 message with a separator string of '|^~\&#' to the /reports or /waters endpoint
2. Verify it does not break

## Changes
- Replace '|^~\&#' with '|^~\&' on incoming hl7 files.
- Add tests
